### PR TITLE
CB-17837 - Roll forward sync fails because some 'p' ending cloudera-m…

### PR DESCRIPTION
**Pull request by pdarvasi**
…anager-server versions are named differently in repository and image catalog

This commit:
 - introduces a package pattern that removes unwanted characters from
end of pattern

See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/resources/application.yml [hunk 1]
```diff
@@ -792,9 +792,11 @@ cb:
       - name: cm
         pkg:
           - name: cloudera-manager-agent
-            pattern: "(.*)-(.*)\\..*"
+            pattern: "(.*)-([0-9]+)[a-zA-z]*\\..*"
+            validateForMultipleVersions: true
```
Comment (nadamdb) [inline core/src/main/resources/application.yml:796]: Wouldn't object creation from config fail if the new field is added here but the PackageName class is not extended with the new field? Or is it just ignored?

**Review by nadamdb — APPROVED** [review]
LGTM

**Review by keyki — APPROVED** [review]

**Review by keyki — APPROVED** [review]

**Review by keyki — APPROVED** [review]

**Review by nadamdb — APPROVED** [review]

**Comment by lturcsanyi** [comment]
is this really required to merge into 2.60?

**Comment by keyki** [comment]
> is this really required to merge into 2.60?

Yes, we intend to do a hotfix. It is causing upgrade failures once retry is used. We've seen several failures in prod already.

**Review by lturcsanyi — APPROVED** [review]

**Pull request merged**